spec/rubocop/cop/performance/count_spec.rb

```diff
@@ -86,6 +86,20 @@
       expect(cop.highlights).to eq(["#{selector}(&:something).count"])
     end
 
+    it "registers an offense for #{selector}(&:something).count " \
+       'called on a class that includes or extends an enumerable' do
+      source = ['class A < Array',
+                '  def count(&block)',
+                "    #{selector}(&block).count",
+                '  end',
+                'end']
+      inspect_source(cop, source)
+
+      expect(cop.messages)
+        .to eq(["Use `count` instead of `#{selector}...count`."])
```

Comment:
The main confusion here would be caused by someone defining `count` to use `select..count` which itself seems weird and should be unnecessary. The auto-correct here would result in a recursive call. We could probably check for a method that calls itself with the same params, but this seems like an edge case. I mentioned this to @bquorning in #2080. If the outer method was named anything else, this wouldn't be an issue.